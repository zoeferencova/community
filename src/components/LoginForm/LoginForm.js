import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { ButtonLight, ButtonDark, Input, Label, Error } from "../Utils/Utils";
import AuthApiService from "../../services/auth-api-service";

import styles from "./LoginForm.module.css";

class LoginForm extends Component {
    state = { 
        error: null,
        loading: false
    }

    handleSubmit(e) {
        e.preventDefault();

        this.setState({...this.state, loading: true })

        const { email, password } = e.target;

        AuthApiService.postLogin({
            email: email.value,
            password: password.value,
        })
            .then(user => {
                email.value = "";
                password.value = "";
                this.setState({...this.state, loading: false })
                !user.location ? this.props.history.push("/location") : this.props.history.push("/home");
            })
            .then(res => this.props.setLoggedIn(true))
            .catch(res => {
                this.setState({...this.state, error: res.error })
            })
    }
    
    render() {
        return (
            <form onSubmit={e => this.handleSubmit(e)} className={styles.form}>
                {this.state.error && <Error message={this.state.error} />}                
                <Label htmlFor="email">Email</Label>
                <Input required type="text" name="email" id="email" />
                <Label htmlFor="password">Password</Label>
                <Input required type="password" name="password" id="password" />
                <div className={styles.buttonContainer}>
                    <ButtonLight type="button" onClick={() => this.props.history.push("/")}>Cancel</ButtonLight>
                    <ButtonDark type="submit" loading={this.state.loading.toString()}>Sign in</ButtonDark>
                </div>
                
            </form>
        )
    }
}

export default withRouter(LoginForm)