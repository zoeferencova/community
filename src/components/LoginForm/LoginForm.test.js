import React from "react";
import { render } from "enzyme";
import LoginForm from "./LoginForm";
import { BrowserRouter } from "react-router-dom";

describe("LoginForm", () => {
    it("renders LoginForm component", () => {
        const wrapper = render(
            <BrowserRouter>
                <LoginForm />
            </BrowserRouter>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});