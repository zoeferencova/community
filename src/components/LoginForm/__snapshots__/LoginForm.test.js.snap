// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders LoginForm component 1`] = `
<form
  class="form"
>
  <label
    class="label "
    for="email"
  >
    Email
  </label>
  <input
    class="input "
    id="email"
    name="email"
    required=""
    type="text"
  />
  <label
    class="label "
    for="password"
  >
    Password
  </label>
  <input
    class="input "
    id="password"
    name="password"
    required=""
    type="password"
  />
  <div
    class="buttonContainer"
  >
    <button
      class="button "
      type="button"
    >
      Cancel
    </button>
    <button
      class="buttonDark "
      loading="false"
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;
